src/transcripts/david-k-piano.md

```diff
@@ -6,3 +6,207 @@ location: Q&A Channel Reactiflux
+
+Another reason to choose XState is to communicate app logic with the rest of your team, even if they’re not as technical - designers, PMs, etc. XState allows you to visualize and simulate your app logic.
+
+---
+
+<a name="is-there-another-design-pattern" href="#is-there-another-design-pattern"># </a>**Q:** Is there another design pattern apart from state machines that would provide complete control over the system? – amaankulshreshtha
+
+I mean, … sorta? There’s petri nets 😂 But anyway, the question is not "should I use state machines or not?" because you already are, technically. Rather, the question is "should I make my state machines explicit, or keep them implicit?"
+
+---
+
+<a name="do-you-have-a-rule" href="#do-you-have-a-rule"># </a>**Q:** Do you have a rule of thumb for when the complexity in code starts to merit using state machines to model it? – vcarl
+
+My rule of thumb is when I have more than 2 states and more than 2 events. If you start seeing this…
+
+```ts
+const [isFoo, setIsFoo] = useState(…);
+const [isBar, setIsBar] = useState(…);
+const [isBaz, setIsBaz] = useState(…);
+// …
+```
+
+then please please just use a state machine.
+
+---
+
+<a name="i-think-a-lot-of" href="#i-think-a-lot-of"># </a>**Q:** I think a lot of more Jr. folks (myself included) have a hard time envisioning the journey from a blank slate to something like XState. What did your development journey look like? Any major roadblocks or difficulties? – DepartureAM
+
+Do you mean creating a library like XState, or using XState in an application?
+amaankulshreshtha — Today at 4:53 PM
+[Q&Q] I used XState do help me "prototype" what my system was going to look like. Suppose, I was happy with my machine and wanted to deploy it (the project got deported before I got there), is it possible to get a real diagram like this once it’s out there in production? If so… how exactly can one see this? Or is it just for prototyping purposes?
+https://cdn.discordapp.com/attachments/193117606629081089/951598768166404126/Screen_Shot_2022-03-10_at_4.51.01_PM.png
+
+## With the VS Code extension, and some of the future tools we have planned, our goal is to have your diagrams from state machines be ✨ living diagrams ✨ that stay up-to-date with your code. So yes, this is possible and we’ll be adding even
+
+<a name="can-we-get-a-vscode" href="#can-we-get-a-vscode"># </a>**Q:** Can we get a vscode extension to visualize a state machine if we write the code for it ? – more features
+
+Yes! And @dilraj this answers your question too: https://marketplace.visualstudio.com/items?itemName=statelyai.stately-vscode
+dilraj — Today at 4:53 PM
```

Comment:
The order seems a bit messed up. Also the heading 2 on `line 207`

Reply:
Should be fixed, Discord changed up how messages copy out and it threw off my muscle memory haha